lighthouse-core/lib/network-recorder.js

```diff
@@ -327,6 +327,37 @@ class NetworkRecorder extends EventEmitter {
     return request;
   }
 
+  /**
+   * @param {NetworkRequest} record The record to find the initiator of
+   * @param {Map<string, NetworkRequest>} Network requests that are possibly prefetch requests,
+   *     keyed by URL.
+   * @param {Map<string, NetworkRequest>} Network requests that are definitely not prefetch
+   *     requests, keyed by URL.
+   * @param {?NetworkRequest} The request that initiated record, or null if no valid initiator
+   *     was found.
+   *@private
+   */
+  static _chooseInitiatorRequest(record, possiblePrefetchRecords, nonPrefetchRecords) {
+    if (record.redirectSource && record.redirectSource.responseReceivedTime <= record.startTime) {
```

Comment:
IMO this one seems less necessary to have the check on and I'd be tad concerned about weird timing protocol issues making these a few ms off or something and missing the request, but I don't feel super strongly